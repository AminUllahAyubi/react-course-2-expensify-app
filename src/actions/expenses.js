
import uuid from "uuid"
// ADD_EXPENSE
export const addExpense=(
    {
        id=uuid(),
        description='',
        note='',
        amount=0,
        createAt=0
    
    }={})=>{
    return{
        type:"ADD_EXPENSE",
        expense:{
            id:uuid(),
            description,
            note,
            amount,
            createAt
        }
    }
}

// REMVOE_EXPENSE
export const removeExpense=({id})=>{
    return{
        type:'REMOVE_EXPENSE',
        id
    }
}
// EDITE_EXPENSE
export const editeExpense=(id,update)=>{
    return{
        type:'EDITE_EXPENSE',
        id,
        update
    }
}

