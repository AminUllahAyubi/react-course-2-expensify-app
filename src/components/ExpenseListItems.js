import React, { createElement } from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
export default ({ id, description, note, amount, createAt}) => {
    return (
        <div>
            <h3>
                <Link to={`/edit#${id}`}>{description}</Link></h3>
            <p>{amount} - {createAt}</p>
        </div>
    )
}
// const mapStateToState = (state) => {
//     return {
//         expenses: state.expenses
//     }
// }
// export default connect(mapStateToState)(ExpenseListItems);
// export default ExpenseListItems;

