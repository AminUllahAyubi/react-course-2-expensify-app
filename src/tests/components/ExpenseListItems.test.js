import React from "react";
import { shallow } from "enzyme";
import expenses from "../fixtures/expenses";
import ExpenseListItems from "../../components/ExpenseListItems";
test('should test the expensen list items',()=>{
    const wrapper=shallow(<ExpenseListItems {...expenses[0]}/>);
    expect(wrapper).toMatchSnapshot();
})
