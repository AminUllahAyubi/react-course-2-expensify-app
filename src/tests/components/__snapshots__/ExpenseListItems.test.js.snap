// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test the expensen list items 1`] = `
<div>
  <h3>
    <Link
      replace={false}
      to="/edit#1"
    >
      Water
    </Link>
  </h3>
  <p>
    90
     - 
    0
  </p>
</div>
`;
